File: Python/Prodi.py
from Mahasiswa import Mahasiswa
from Dosen import Dosen
from Course import Course

class Prodi :
    # Deklarasi variabel
    __namaProdi = ""
    __kode = ""
    __mahasiswa = []
    __dosen = []
    __course = []
    
    # Konstruktor
    # Penggunaan list disini dikarenakan untuk menyimpan data yang memiliki hubungan composite dengan Prodi    
    def __init__(self, mahasiswa = [], dosen = [], course = [],namaProdi = "", kode = ""):
        self.__namaProdi = namaProdi
        self.__kode = kode
        self.__mahasiswa = mahasiswa
        self.__dosen = dosen
        self.__course = course
    
    # Get
    def get_namaProdi(self):
        return self.__namaProdi
    def get_kode(self):
        return self.__kode
    
    # Get dari hubungan composite, menggunakan loop karena agar dapat mengambil semua variabel yang ada di kelas mahasiswa
    def get_mahasiswa(self):
        print("    DATA MAHASISWA")
        for index, i in enumerate(self.__mahasiswa) :
            print(f"\tMahasiswa ke-{index+1}")
            print("\t\tNIK              :",i.get_nik())
            print("\t\tNama Mahasiswa   :",i.get_nama())
            print("\t\tGender           :",i.get_jenis_kelamin())
            print("\t\tNIM              :",i.get_nim())
            print("\t\tFakultas         :",i.get_fakultas())
            print("\t\tAsal Univ        :", i.get_asalUniv())
            print("\t\tEmail            :", i.get_emailEdu())
    
    # Get dari hubungan composite, menggunakan loop karena agar dapat mengambil semua variabel yang ada di kelas dosen       
    def get_dosen(self):
        print("    DATA DOSEN")
        for index, i in enumerate(self.__dosen) :
            print(f"\tDosen ke-{index+1}")
            print("\t\tNIK              :",i.get_nik())
            print("\t\tNama Dosen       :",i.get_nama())
            print("\t\tGender           :",i.get_jenis_kelamin())
            print("\t\tNIP              :",i.get_nip())
            print("\t\tFakultas         :",i.get_fakultas())
            print("\t\tPendidikan Akhir :",i.get_pendidikanAkhir())
            print("\t\tKeahlian         :",i.get_keahlian())
            print("\t\tAsal Univ        :", i.get_asalUniv())
            print("\t\tEmail            :", i.get_emailEdu())

    # Get dari hubungan composite, menggunakan loop karena agar dapat mengambil semua variabel yang ada di kelas course
    def get_course(self):
        print("    DATA MATKUL")
        for index, i in enumerate(self.__course) :
            print(f"\tCourse ke-{index+1}")
            print("\t\tNama Matkul      :",i.get_namaMatkul())

File: C++/Prodi.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

class Prodi {
    private:
        string namaProdi;
        string kode;
        vector<Mahasiswa> mahasiswa;
        vector<Dosen> dosen;
        vector<Course> course;

    public:
        // Konstruktor
        // Penggunaan list disini dikarenakan untuk menyimpan data yang memiliki hubungan composite dengan Prodi
        Prodi(vector<Mahasiswa> mahasiswa = {}, vector<Dosen> dosen = {}, vector<Course> course = {}, string namaProdi = "", string kode = "") {
            this->namaProdi = namaProdi;
            this->kode = kode;
            this->mahasiswa = mahasiswa;
            this->dosen = dosen;
            this->course = course;
        }

        // Getter
        string getNamaProdi(){
            return this->namaProdi;
        }
        string getKode(){
            return this->kode;
        }
        // get mahasiswa untuk menampilkan data semua mahasiswa
        void get_mahasiswa(){
            cout <<"    DATA MAHASISWA" <<endl;
            for(Mahasiswa cek:mahasiswa){
                cout << "\t\tNIK              :" << cek.getNik() <<endl;
                cout << "\t\tNama Mahasiswa   :" << cek.getNama() <<endl;
                cout << "\t\tGender           :" << cek.getJenisKelamin() <<endl;
                cout << "\t\tNIM              :" << cek.getNim() <<endl;
                cout << "\t\tFakultas         :" << cek.getFakultas()<<endl;
                cout << "\t\tAsal Univ        :" << cek.getAsalUniv() <<endl;
                cout << "\t\tEmail            :" << cek.getEmailEdu() <<endl;
            }
        }
        // get dosen untuk menampilkan data semua dosen
        void get_dosen(){
            cout <<"    DATA DOSEN" <<endl;
            for(Dosen cek:dosen){
                cout << "\t\tNIK                :" << cek.getNik() <<endl;
                cout << "\t\tNama Dosen         :" << cek.getNama() <<endl;
                cout << "\t\tGender             :" << cek.getJenisKelamin() <<endl;
                cout << "\t\tNIP                :" << cek.getNip() <<endl;
                cout << "\t\tFakultas           :" << cek.getFakultas()<<endl;
                cout << "\t\tKeahlian           :" << cek.getKeahlian()<<endl;
                cout << "\t\tPendidikan Akhir   :" << cek.getPendidikanAkhir()<<endl;
                cout << "\t\tAsal Univ          :" << cek.getAsalUniv() <<endl;
                cout << "\t\tEmail              :" << cek.getEmailEdu() <<endl;
            }
        }
        // get course untuk menampilkan data semua course
        void get_course(){
            cout <<"    DATA MATKUL" <<endl;
            for(Course cek:course){
                cout << "\t\tNama Matkul        :" << cek.getMatkul() <<endl;
            }
        }
        
        // Destruktor
        ~Prodi(){}
};

